consensus/core/src/round_tracker.rs

```diff
@@ -59,27 +55,33 @@ impl PeerRoundTracker {
         }
     }
 
-    /// Update accepted rounds based on a new block created locally or received from the network
-    /// and its excluded ancestors
-    pub(crate) fn update_from_accepted_block(&mut self, extended_block: &ExtendedBlock) {
-        let block = &extended_block.block;
-        let excluded_ancestors = &extended_block.excluded_ancestors;
-        let author = block.author();
-
-        // Update author accepted round from block round
-        self.block_accepted_rounds[author][author] =
-            self.block_accepted_rounds[author][author].max(block.round());
-
-        // Update accepted rounds from included ancestors
-        for ancestor in block.ancestors() {
-            self.block_accepted_rounds[author][ancestor.author] =
-                self.block_accepted_rounds[author][ancestor.author].max(ancestor.round);
+    /// Update accepted rounds based on a new block created locally, received from the
+    /// network, commit sync, or via live sync.
+    pub(crate) fn update_from_accepted_blocks(&mut self, blocks: &Vec<VerifiedBlock>) {
+        for block in blocks {
+            let author = block.author();
+
+            // Update author accepted round from block round
+            self.block_accepted_rounds[author][author] =
```

Comment:
How about updating `block_accepted_rounds[own_index][block author]` here as well, so you don't need to call `update_from_excluded_ancestors()` on own proposed block?

Also, add comment to mention that even though most of the logic in `update_from_accepted_blocks()` can run earlier in block handler and synchronizer, they run here instead for simplicity.